code/lib/components/src/icon/icon.tsx

```diff
@@ -1,39 +1,54 @@
-import React, { ComponentProps, memo } from 'react';
+import React, { FunctionComponent, ComponentProps, memo } from 'react';
 import { styled } from '@storybook/theming';
-import icons, { IconKey } from './icons';
+import { icons, IconKey } from './icons';
 
-import Svg from './svg';
+const Svg = styled.svg`
+  display: inline-block;
+  shape-rendering: inherit;
+  transform: translate3d(0, 0, 0);
+  vertical-align: middle;
 
-const Path = styled.path({
-  fill: 'currentColor',
-});
+  path {
+    fill: currentColor;
+  }
+`;
 
-export interface IconsProps extends ComponentProps<typeof Svg> {
-  icon?: IconKey;
-  symbol?: IconKey;
+/**
+ * An Icon is a piece of visual element, but we must ensure its accessibility while using it.
+ * It can have 2 purposes:
+ *
+ * - *decorative only*: for example, it illustrates a label next to it. We must ensure that it is ignored by screen readers, by setting `aria-hidden` attribute (ex: `<Icon icon="check" aria-hidden />`)
+ * - *non-decorative*: it means that it delivers information. For example, an icon as only child in a button. The meaning can be obvious visually, but it must have a proper text alternative via `aria-label` for screen readers. (ex: `<Icon icon="print" aria-label="Print this document" />`)
+ */
+export interface IconsProps {
+  icon?: IconType;
+  symbol?: IconType;
 }
 
-// TODO: if we can resize the 1024 to 20, we can remove the size attributes
-export const Icons = memo<IconsProps>(({ icon, symbol, ...props }) => (
-  <Svg viewBox="0 0 1024 1024" {...props}>
-    {symbol ? <use xlinkHref={`#icon--${symbol}`} /> : <Path d={icons[icon]} />}
-  </Svg>
-));
+export const Icons: FunctionComponent<IconsProps> = ({ icon, symbol, ...props }: IconsProps) => {
+  return (
+    <Svg viewBox="0 0 14 14" width="14px" height="14px" {...props}>
+      {symbol ? <use xlinkHref={`#icon--${symbol}`} /> : icons[icon]}
+    </Svg>
+  );
+};
+
+export type IconType = keyof typeof icons;
 
 export interface SymbolsProps extends ComponentProps<typeof Svg> {
   icons?: IconKey[];
 }
 
 export const Symbols = memo<SymbolsProps>(({ icons: keys = Object.keys(icons) }) => (
```

Comment:
This component differed from the one in the Design System in that there was an effort to make a symbol instead of using SVGs directly (sort of like a sprite). I left that functionality in tact here, but I am unsure if it is necessary. Thoughts?